code/game/objects/items.dm

```diff
@@ -337,6 +346,34 @@ GLOBAL_VAR_INIT(embedpocalypse, FALSE) // if true, all items will be able to emb
 	research_msg += "."
 	. += research_msg.Join()
 
+/**
+ *
+ * Details the stats of the examined weapon
+ *
+ * This function is called when the user clicks the hyperlink in
+ * /obj/item/examine. This function simply compiles some relevant
+ * weapon data and outputs it to the user.
+ * Arguments:
+ *  * href - Unused
+ *  * href-list - List provided by the href of input values, used to know what hyperlinked action is being attempted
+ */
+
+/obj/item/Topic(href, href_list)
+	if(href_list["examine"])
+		var/readout = "" /// Readout is used to store the text block output to the user so it all can be sent in one message
+		/// Override to disable standard outputs for items that dont need it. See: Ammo
+		if(!note_override)
+			readout += "DMG: [force]\n" /// Standard attack damage
+			readout += "THRW: [throwforce]\n" /// Throwing damage
+			readout += uppertext("TYPE: [damtype]\n") /// Damage type
+			readout += "AP: [armour_penetration]%\n" /// Armour pen
+			readout += "BLOC: [block_chance]%\n" /// Block Chance
+		/// Add on the offensive notes
+		if(offensive_notes)
+			readout += offensive_notes
+		to_chat(usr, "<span class='notice'>[readout]</span>")
```

Comment:
instead of pre-creating offensive_notes, don't store it at all and generate it here. Set readout to the output of some proc, in a similar fashion to how examine() works, and use that proc to generate the text